sdv/multi_table/base.py

```diff
@@ -85,26 +87,31 @@ def get_metadata(self):
         return self.metadata
 
     def _validate_foreign_keys(self, data):
+        error_msg = None
         errors = []
         for relation in self.metadata._relationships:
-            child_table = data[relation['child_table_name']]
-            parent_table = data[relation['parent_table_name']]
-            child_row = child_table[relation['child_foreign_key']]
-            parent_row = parent_table[relation['parent_primary_key']]
-            missing_values = child_row[~child_row.isin(parent_row)].unique()
-            if any(missing_values):
-                message = ', '.join(missing_values[:5].astype(str))
-                if len(missing_values) > 5:
-                    message = f'({message}, + more).'
-                else:
-                    message = f'({message}).'
-
-                errors.append(
-                    f"Error: foreign key column '{relation['child_foreign_key']}' contains "
-                    f'unknown references: {message}'
-                )
-
-        return errors
+            child_table = data.get(relation['child_table_name'])
+            parent_table = data.get(relation['parent_table_name'])
+            if isinstance(child_table, pd.DataFrame) and isinstance(parent_table, pd.DataFrame):
```

Comment:
why did we add the check for if they're dataframes?

Reply:
Before I was raising the error if the check failed on the `SingleTableSynthesizer` , however, since we want to have all the errors raised at once, I have to check this otherwise it will fail as key error or something else.